src/Sylius/Bundle/PayumBundle/Action/ConvertPaymentToBe2BillAction.php

```diff
@@ -43,6 +46,18 @@ protected function composeDetails(PaymentInterface $payment, TokenInterface $tok
         $details['DESCRIPTION'] = sprintf('Order containing %d items for a total of %01.2f', $order->getItems()->count(), $order->getTotal() / 100);
         $details['ORDERID'] = $payment->getId();
 
-        $payment->setDetails($details);
+        $request->setResult($details);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function supports($request)
+    {
+        return
+            $request instanceof Convert &&
+            $request->getSource() instanceof PaymentInterface &&
+            $request->getTo() == 'array'
```

Comment:
@stloyd you dont have to write strick here and there, if we agree on it here, I change it in all places.